classes/class-dintero-checkout-order-management.php

```diff
@@ -221,7 +223,20 @@ public function cancel_order( $order_id ) {
 			return;
 		}
 
-		// Check if Dintero order has been canceled in the back-office.
+		// If the payment method is Swish, we need to refund instead of cancel.
+		if ( 'swish' === strtolower( $payment_method ) ) {
+			$refund = wc_create_refund(
+				array(
+					'amount'         => $order->get_total(),
+					'reason'         => 'Swish order canceled by customer.',
+					'order_id'       => $order_id,
+					'refund_payment' => true,
+				)
+			);
+			return;
+		}
```

Comment:
Issue/Question: I don't think we should be creating an actual refund in WooCommerce for this since that might effect other things as well. Rather i think we should instead trigger the refund request instead when this happens. Or have you discussed this with Niklas?